prime-dotnet-webapi/Services/CommunitySiteService.cs

```diff
@@ -124,16 +124,18 @@ public async Task<PermissionsRecord> GetPermissionsRecordAsync(int siteId)
 
         private void UpdateAddress(Site current, CommunitySiteUpdateModel updated)
         {
-            if (updated?.PhysicalAddress != null)
+            if (updated.PhysicalAddress == null)
             {
-                if (current.PhysicalAddress == null)
-                {
-                    current.PhysicalAddress = updated.PhysicalAddress;
-                }
-                else
-                {
-                    _context.Entry(current.PhysicalAddress).CurrentValues.SetValues(updated.PhysicalAddress);
-                }
+                return;
+            }
+
+            if (current.PhysicalAddress == null)
+            {
+                current.PhysicalAddress = updated.PhysicalAddress;
```

Comment:
I think you might have to do a `_context.Addresses.Remove(current.PhysicalAddress)` here, but I'm not certain.